src/tests/zfs/zfs_volume_snapshot/zfs_volume_snapshot_test.go

```diff
@@ -0,0 +1,170 @@
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openebs/openebs-e2e/common/e2e_config"
+	"github.com/openebs/openebs-e2e/common/e2e_ginkgo"
+	"github.com/openebs/openebs-e2e/common/mayastor/snapshot"
+	"github.com/openebs/openebs-e2e/common/zfs"
+
+	"github.com/openebs/openebs-e2e/common"
+	"github.com/openebs/openebs-e2e/common/k8stest"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	coreV1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+/*
+   Background:
+        Given a k8s cluster is running with the product installed
+        And snapshot CRDs exists in k8s cluster
+
+   Scenario: Snapshot creation for zfs volume
+        Given a volume has been successfully created
+        When snapshot is created for the volume
+        Then the snapshot should be successfully created
+		And the snapshot  object should be ready
+        And the snapshot content object associated with snapshot should be ready
+*/
+
+var nodeConfig zfs.ZfsNodesDevicePoolConfig
+var app k8stest.FioApplication
+var snapshotClassName, snapshotName, snapshotNamespace string
+
+func volumeSnapshotTest(decor string, engine common.OpenEbsEngine, volType common.VolumeType, fstype common.FileSystemType, volBindModeWait bool) {
+	app = k8stest.FioApplication{
+		Decor:                   decor,
+		VolSizeMb:               2048,
+		OpenEbsEngine:           engine,
+		VolType:                 volType,
+		FsType:                  fstype,
+		Loops:                   3,
+		VolWaitForFirstConsumer: volBindModeWait,
+	}
+
+	// setup sc parameters
+	app.Zfs = k8stest.ZfsOptions{
+		PoolName:      nodeConfig.PoolName,
+		ThinProvision: common.No,
+		RecordSize:    "128k",
+		Compression:   common.Off,
+		DedUp:         common.Off,
+	}
+
+	if app.FsType == common.ZfsFsType || app.FsType == common.BtrfsFsType {
+		app.FsPercent = 50
+	}
+
+	logf.Log.Info("create sc, pvc, fio pod")
+	err := app.DeployApplication()
+	Expect(err).ToNot(HaveOccurred())
+
+	logf.Log.Info("Sleep for 30 seconds before taking snapshot")
+	time.Sleep(30 * time.Second)
+
+	// snapshot steps
+	snapshotClassName = fmt.Sprintf("snapshotclass-%s", app.GetPvcName())
+	snapshotName = fmt.Sprintf("snapshot-%s", app.GetPvcName())
+	snapshotNamespace = common.NSDefault
+	logf.Log.Info("Create Snapshot", "Snapshot class", snapshotClassName, "Snapshot", snapshotName, "Namespace", snapshotNamespace)
+	csiDriver := e2e_config.GetConfig().Product.ZfsEngineProvisioner
+	// create snapshot for volume
+	snapshotObj, snapshotContentName, err := k8stest.CreateVolumeSnapshot(snapshotClassName, snapshotName, app.GetPvcName(), common.NSDefault, csiDriver)
+	Expect(err).ToNot(HaveOccurred())
+	logf.Log.Info("Snapshot Created ", "Snapshot", snapshotObj, "Snapshot Content Name", snapshotContentName)
+
+	Expect(snapshotContentName).ToNot(BeEmpty(), "snapshot content name should not be empty for snapshot %s", snapshotName)
+	Expect(snapshotObj).ShouldNot(BeNil())
+
+	// verify Snapshot CR
```

Comment:
nit: CR should be creation?

Reply:
updated the comment https://github.com/openebs/openebs-e2e/compare/cd57132f0bb233bb1a597c6e7fff62485261941c..391fb496819c131f270881e45e1d570a86ff37a6